code/modules/vehicles/attachments.dm

```diff
@@ -0,0 +1,148 @@
+	max_rounds = 1000
+	point_cost = 10
+
+/obj/item/ammo_magazine/tank/tank_glauncher
+	name = "Grenade Launcher Magazine"
+	desc = "A secondary armament grenade magazine"
+	caliber = "grenade"
+	icon_state = "glauncher_2"
+	w_class = 9
+	default_ammo = /datum/ammo/grenade_container
+	max_rounds = 10
+	point_cost = 25
+
+/obj/item/ammo_magazine/tank/tank_glauncher/update_icon()
+	if(current_rounds >= max_rounds)
+		icon_state = "glauncher_2"
+	else if(current_rounds <= 0)
+		icon_state = "glauncher_0"
+	else
+		icon_state = "glauncher_1"
+
+
+/obj/item/ammo_magazine/tank/tank_slauncher
+	name = "Smoke Launcher Magazine"
+	desc = "A support armament grenade magazine"
+	caliber = "grenade"
+	icon_state = "slauncher_1"
+	w_class = 12
+	default_ammo = /datum/ammo/grenade_container/smoke
+	max_rounds = 6
+	point_cost = 5
+
+/obj/item/ammo_magazine/tank/tank_slauncher/update_icon()
+	icon_state = "slauncher_[current_rounds <= 0 ? "0" : "1"]"
+
+///////////////
+// AMMO MAGS // END
+///////////////
+
+/obj/vehicle/tank/attackby(obj/item/I, mob/user) //This handles reloading weapons, or changing what kind of mags they'll accept. You can have a passenger do this
+	if(user.loc == src) //Stops safe healing
+		to_chat(user, "<span class='warning'>You can't reach [src]'s hardpoints while youre seated in it.</span>")
+		return
+	if(iswelder(I)) //Weld to repair the tank
+		if(obj_integrity >= max_integrity)
+			to_chat(user, "<span class='warning'>You can't see any visible dents on [src].</span>")
+			return
+		var/obj/item/tool/weldingtool/WT = I
+		if(!WT.isOn())
+			to_chat(user, "<span class='warning'>You need to light your [WT] first.</span>")
+			return
+		if(!do_after(user, 20 SECONDS, TRUE, src, BUSY_ICON_BUILD, extra_checks = iswelder(I) ? CALLBACK(I, /obj/item/tool/weldingtool/proc/isOn) : null))
+			user.visible_message("<span class='notice'>[user] stops repairing [src].</span>",
+				"<span class='notice'>You stop repairing [src].</span>")
+			return
+		WT.remove_fuel(3, user) //3 Welding fuel to repair the tank. To repair a small tank, it'd take 4 goes AKA 12 welder fuel and 1 minute
+		obj_integrity += 100
+		if(obj_integrity > max_integrity)
+			obj_integrity = max_integrity //Prevent overheal
+		to_chat(user, "<span class='warning'>You weld out a few of the dents on [src].</span>")
+		update_icon() //Check damage overlays
+		return
+	if(is_type_in_list(I, primary_weapon?.accepted_ammo))
+		var/mob/living/M = user
+		var/time = 8 SECONDS - (1 SECONDS * M.mind.cm_skills.large_vehicle)
+		to_chat(user, "You start to swap out [primary_weapon]'s magazine...")
+		if(do_after(M, time, TRUE, src, BUSY_ICON_BUILD))
+			playsound(get_turf(src), 'sound/weapons/working_the_bolt.ogg', 100, 1)
+			primary_weapon.ammo.forceMove(get_turf(user))
+			primary_weapon.ammo = I
+			to_chat(user, "You load [I] into [primary_weapon] with a satisfying click.")
+			user.transferItemToLoc(I,src)
+		return
+	if(is_type_in_list(I, secondary_weapon?.accepted_ammo))
```

Comment:
You can write this so you don't need duplicate code blocks. Really you need to break apart the code in this proc in general, you have too much happening here